### PR TITLE
Delete classic mode install/uninstall/upgrade commands

### PR DESCRIPTION
**Pull request by michi-covalent**
Ref: #2327

### CODE REVIEW

File: internal/cli/cmd/cmd.go [hunk 1]
```diff
@@ -88,20 +87,10 @@ cilium connectivity test`,
 		newCmdStatus(),
 		newCmdSysdump(hooks),
 		newCmdVersion(),
+		newCmdInstallWithHelm(),
+		newCmdUninstallWithHelm(),
+		newCmdUpgradeWithHelm(),
 	)
-	if utils.IsInHelmMode() {
```
Comment (gandro) [inline internal/cli/cmd/cmd.go:92]: I think we need to update the README as well? https://github.com/cilium/cilium-cli?tab=readme-ov-file#helm-installation-mode

**Review by gandro — CHANGES_REQUESTED** [review]
Overall this looks good to me, modulo some doc changes.

Would it help to do a cilium/cilium CI run with a version of cilium-cli with this patch applied? Or are we certain that the classic mode is not used anywhere?

File: internal/cli/cmd/cmd.go (hunk 1 above)
Comment (michi-covalent) [reply internal/cli/cmd/cmd.go:92]: yeah i was going to update README.md in a subsequent PR after removing the classic mode from all the commands but i don't mind updating it in this PR.

i'm pretty sure the classic mode is not being used in cilium/cilium CI anymore, but i'll run the CI to make sure 🚀 🙏

**Comment by michi-covalent** [comment]
running the CI here => https://github.com/cilium/cilium/pull/30978

**Pull request merged**